addons/block_code/ui/block_canvas/node_block_canvas/node_block_canvas.gd

```diff
@@ -21,9 +21,12 @@ func generate_script_from_current_window(script_inherits: String = ""):
 
 	var script: String = ""
 
-	script += "extends %s\n\n" % script_inherits
+	script += "extends %s\n\n" % bsd.script_inherits
 
-	script += "var VAR_DICT := {}\n\n"
+	for variable in bsd.variables:
+		script += "var %s\n\n" % variable.var_name.to_snake_case()
```

Comment:
Is it possible to declare these as typed variables? One advantage of typed variables is that they're [initialized to a default] instead of `null`.

Reply:
Yea, it would be nice to have defaults in there somehow, I was thinking about how we would accomplish that. At the moment we have no way to convert from a `Variant.Type` to the text Godot uses to specify types, but we could definitely have another dictionary in `types.gd` that maps from a `Variant.Type` to a default value. I'll take a look at that.

Reply:
Doesn't [type_string](https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#class-globalscope-method-type-string) do that?